Engine/TransactionHandlers/BrokerageTransactionHandler.cs

```diff
@@ -76,23 +89,29 @@ public class BrokerageTransactionHandler : ITransactionHandler
         /// </summary>
         public ConcurrentDictionary<int, Order> Orders
         {
-            get { return _orders; }
+            get {
+                return new ConcurrentDictionary<int, Order>(_closedOrders.Concat(_openOrders).
+                                                              ToDictionary(kvp => kvp.Key, kvp => kvp.Value));
+            }
         }
 
         /// <summary>
         /// Gets the permanent storage for all order tickets
         /// </summary>
         public ConcurrentDictionary<int, OrderTicket> OrderTickets
         {
-            get { return _orderTickets; }
+            get {
+                return new ConcurrentDictionary<int, OrderTicket>(_closedOrderTickets.Concat(_openOrderTickets).
+                                                                  ToDictionary(kvp => kvp.Key, kvp => kvp.Value));
+            }
         }
 
         /// <summary>
         /// Gets the current number of orders that have been processed
         /// </summary>
         public int OrdersCount
         {
-            get { return _orders.Count; }
+            get { return _closedOrders.Count + _openOrders.Count; }
```

Comment:
I recall a discussion about using the current order id as the orders count to prevent reads on concurrent dictionary's count property. Should we use that here?

Reply:
Your right. Returning `LastOrderId` for calls to `OrdersCount` at `SecurityTransactionManager` means this is not required anymore, so I'll remove it. What do you think?